Makefile

```diff
@@ -52,7 +52,7 @@ LDFLAGS += -lpthread
 ifneq ($(MODE),debug)
 	CFLAGS += -O3 -DNDEBUG
 else
-	CFLAGS += -g
+	CFLAGS += -O0 -g
```

Comment:
GCC allows us to use -g with -O.
But If we are not using some other optimization option, it consider using -Og with -g.
So I think this commit change is not necessary.